### PR TITLE
Export props along with components

### PR DESCRIPTION
**Pull request by jrparish**
This exposes the prop types in the barrel file so that imports are a little cleaner when extending components.

Before:
```ts
import { Button } from 'react-native-magnus';
import { ButtonProps } from 'react-native-magnus/lib/typescript/src/ui/button/button.type';
```

After:
```ts
import { Button, ButtonProps } from 'react-native-magnus';
```

### CODE REVIEW

File: src/ui/index.ts [hunk 1]
```diff
@@ -1,37 +1,58 @@
...
+export { FabProps } from './fab/fab.type';
 export { Input } from './input/input.component';
+export { InputProps } from './input/input.type';
 export { Mention } from './mention/mention.component';
+export { MentionProps } from './mention/mention.type';
 export { Textarea } from './input/textarea.component';
+export { TextareaProps } from './input/textarea.type';
 export { Div } from './div/div.component';
+export { DivProps } from './div/div.type';
 export { Div as Box } from './div/div.component';
```
Comment (erickriva) [inline src/ui/index.ts:21]: Why change the name to Box?

Comment (jrparish) [reply src/ui/index.ts:21]: It exports both. `Box` is an alias of `Div`

image
https://magnus-ui.com/docs/div/

Comment (erickriva) [reply src/ui/index.ts:21]: Oh sorry, my mistake! Believe I didn't even knew that this alias exists?

**Comment by erickriva** [comment]
one time using that import: `import { Button, ButtonProps } from 'react-native-magnus';`,

`import { ButtonProps } from 'react-native-magnus/lib/typescript/src/ui/button/button.type';`
editor autocomplete will not suggest this, right?

**Comment by jsartisan** [comment]
@jrparish Sorry but can you given an example where you want to use the magnus component's props? 🤔

**Comment by jsartisan** [comment]
> one time using that import: `import { Button, ButtonProps } from 'react-native-magnus';`,
> 
> `import { ButtonProps } from 'react-native-magnus/lib/typescript/src/ui/button/button.type';`
> editor autocomplete will not suggest this, right?

To be frank, I never tried using Magnus components props :P

**Comment by erickriva** [comment]
> To be frank, I never tried using Magnus components props.

I think it's always good to export 'types' together.
If you create a component that need to extend this types, it's there ready to use.
I was thinking about doing a PR like this too, but I didn't know where to start 😄

**Comment by jrparish** [comment]
It's really about extending components as @erickriva mentioned.

Here is one example of a `Link` component that extends the `Text` component. 
image

Another case could be creating field components, maybe you want to create an input field that has a label associated with it. 
Or maybe you want to have a default implementation of a button, but still allow all the properties to be overridden.

**Comment by erickriva** [comment]
> It's really about extending components as @erickriva mentioned.
> 
> Here is one example of a `Link` component that extends the `Text` component. 
> image
> 
> Another case could be creating field components, maybe you want to create an input field that has a label associated with it. 
> Or maybe you want to have a default implementation of a button, but still allow all the properties to be overridden. 
> 

i totally agree with you.

(kind of off-topic) why use Text component instead Button component with bg="transparent"? there's a specific reason for this choice?

**Comment by jrparish** [comment]
> (kind of off-topic) why use Text component instead Button component with bg="transparent"? there's a specific reason for this choice?

I'm not sure, this was just an example I pulled out of our code base 😁 
You're probably right that it should be a button, but same concept applies with the extended props.

**Pull request merged**